checker.cpp

```diff
@@ -1,46 +1,73 @@
 #include <iostream>
-#include <assert.h>
 #include <unistd.h>
+#include <cassert>
+
 using namespace std;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if(temperature > 102 || temperature < 95) {
-    cout << "Temperature critical!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
-    }
+struct Alert {
+    string message;
+    int delayInSeconds;
+};
 
-    return 0;
-  } else if(pulseRate < 60 || pulseRate > 100) {
-    cout << "Pulse Rate is out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+void showAlertWithDelay(const Alert &alert) {
+    cout << alert.message << endl;
+    for (int i = 0; i < 6; i++) {
+        cout << "\r* " << flush;
+        sleep(alert.delayInSeconds);
+        cout << "\r *" << flush;
+        sleep(alert.delayInSeconds);
     }
-    return 0;
-  } else if(spo2 < 90) {
-    cout << "Oxygen Saturation out of range!\n";
-    for (int i = 0; i < 6; i++)
-    {
-      cout << "\r* " << flush;
-      sleep(1);
-      cout << "\r *" << flush;
-      sleep(1);
+}
+
+bool isOutOfRange(float value, float lowerLimit, float upperLimit) {
+    return value < lowerLimit || value > upperLimit;
+}
+
+bool isTemperatureCritical(float temperature) {
+    return isOutOfRange(temperature, 95, 102);
+}
+
+bool isPulseRateOutOfRange(float pulseRate) {
+    return isOutOfRange(pulseRate, 60, 100);
+}
+
+bool isSpo2OutOfRange(float spo2) {
+    return isOutOfRange(spo2, 0, 90);
+}
+
+void checkAndAlert(const Alert &alert, bool condition) {
```

Comment:
Try a better name for the `condition` parameter. What condition does it represent? `outOfRange`?